server/src/operators/card_operator.rs

```diff
@@ -143,202 +143,129 @@ pub async fn search_card_query(
-            search_results: point_ids,
-            total_card_pages: 100,
-        });
-    }
-
-    let mut conn = pool.get().unwrap();
-
-    // SELECT distinct card_metadata.qdrant_point_id, card_collisions.collision_qdrant_id
-    // FROM card_metadata
-    // left outer JOIN card_collisions ON card_metadata.id = card_collisions.card_id
-    // WHERE card_metadata.private = false OR (card_metadata.private = false and card_metadata.qdrant_point_id is null);
-
-    //SELECT DISTINCT "card\_metadata"."qdrant\_point\_id", "card\_collisions"."collision\_qdrant\_id"
-    //FROM ("card\_metadata"
-    //LEFT OUTER JOIN "card\_collisions" ON ("card\_metadata"."id" = "card\_collisions"."card\_id"))
-    //WHERE (("card\_metadata"."private" = $1) OR (("card\_metadata"."private" = $2) AND ("card\_metadata"."qdrant\_point\_id" IS NULL))) -- binds: \[false, false\]
-    use crate::data::schema::card_collisions::dsl as card_collisions_columns;
-    use crate::data::schema::card_metadata::dsl as card_metadata_columns;
-
-    let mut query = card_metadata_columns::card_metadata
-        .left_outer_join(
-            card_collisions_columns::card_collisions
-                .on(card_metadata_columns::id.eq(card_collisions_columns::card_id)),
-        )
-        .select((
-            card_metadata_columns::qdrant_point_id,
-            card_collisions_columns::collision_qdrant_id.nullable(),
-        ))
-        .into_boxed();
+    let mut filter = Filter::default();
+    filter.should.push(Condition::is_empty("private"));
+    filter.should.push(Condition::is_null("private"));
+    filter.should.push(Condition::matches("private", false));
+    filter.should.push(Condition::matches(
+        "authors",
+        current_user_id.unwrap_or_default().to_string(),
+    ));
 
-    query =
-        query
-            .filter(card_metadata_columns::private.eq(false).or(
-                card_metadata_columns::author_id.eq(current_user_id.unwrap_or(uuid::Uuid::nil())),
-            ))
-            .distinct();
     let tag_set_inner = tag_set.unwrap_or_default();
     let link_inner = link.unwrap_or_default();
-    if let Some(tag) = tag_set_inner.get(0) {
-        query = query.filter(card_metadata_columns::tag_set.ilike(format!("%{}%", tag)));
-    }
-    for tag in tag_set_inner.iter().skip(1) {
-        query = query.or_filter(card_metadata_columns::tag_set.ilike(format!("%{}%", tag)));
-    }
 
-    if let Some(link_inner) = link_inner.get(0) {
-        query = query.filter(card_metadata_columns::link.ilike(format!("%{}%", link_inner)));
+    if !tag_set_inner.is_empty() {
+        filter
+            .must
+            .push(Condition::matches("tag_set", tag_set_inner));
     }
-    for link_url in link_inner.iter().skip(1) {
-        query = query.or_filter(card_metadata_columns::link.ilike(format!("%{}%", link_url)));
+
+    if !link_inner.is_empty() {
+        filter.must.push(Condition::matches("link", link_inner));
```

Comment:
Reasonably sure this means the index should be for `index` above

Reply:
Wdym?

Reply:
God dammit, I meant to write `link`